packages/flutter/lib/src/services/text_input.dart

```diff
@@ -896,17 +923,24 @@ class TextInput {
   /// A client that no longer wishes to interact with the text input control
   /// should call [TextInputConnection.close] on the returned
   /// [TextInputConnection].
-  static TextInputConnection attach(TextInputClient client, TextInputConfiguration configuration) {
+  static TextInputConnection attach(TextInputClient client) {
     assert(client != null);
-    assert(configuration != null);
-    assert(_debugEnsureInputActionWorksOnPlatform(configuration.inputAction));
     final TextInputConnection connection = TextInputConnection._(client);
     _clientHandler._currentConnection = connection;
-    SystemChannels.textInput.invokeMethod<void>(
+    _attach(_clientHandler, connection);
+    return connection;
+  }
+
+  static void _attach(TextInputClientHandler handler, TextInputConnection connection) {
+    assert(connection != null);
+    assert(connection._client != null);
+    final TextInputConfiguration configuration = connection._client.createConfiguration();
```

Comment:
To make this less breaking: I wonder if we somehow could hold on to the configuration and just re-use it on re-attach?

Reply:
I'm going to have to do that for the last editing value anyway.  I'm not opposed to it, just means we're passing more state around.  I think this API is cleaner even if breaking.  WDYT?